File: e2e/cypress-javascript/interactions/validators/TaskValidator.js
const { step } = require('@/decorators/step-decorator');
const { TaskInfoModal } = require('@/interactions/pages/TaskInfoModal');

class TaskValidator {
  constructor() {
    this.modal = new TaskInfoModal();
    this.data = step('Validate task info data: {taskData}', this._data.bind(this));
    this.isValid = step('Validate task is marked as valid', this._isValid.bind(this));
    this.isNotValid = step('Validate task is marked as not valid', this._isNotValid.bind(this));
  }

  _data(task) {
    this.modal.title.should('have.text', task.title);
    this.modal.description.should('have.text', task.description);
    this.modal.statusTag(task.status).should('be.visible');
    this.modal.priorityTag(task.priority).should('be.visible');
    return this;
  }

  _isValid() {
    this.modal.validationBadge.should('be.visible');
    return this;
  }

  _isNotValid() {
    this.modal.validationBadge.should('not.exist');
    return this;
  }
}

module.exports = { TaskValidator };

File: e2e/capybara-ruby/interactions/validators/task_validator.rb
# frozen_string_literal: true

require_relative '../../decorators/step_decorator'
require_relative '../../support/capybara_helpers'
require_relative '../pages/task_info_modal'

class TaskValidator
  extend StepDecorator::Methods
  include CapybaraHelpers

  def initialize
    @modal = TaskInfoModal.new
  end

  step 'Validate task info data: {taskData}'
  def data(task)
    expect(@modal.title).to have_text(task[:title])
    expect(@modal.description).to have_text(task[:description])
    expect(@modal.status_tag(task[:status])).to be_visible
    expect(@modal.priority_tag(task[:priority])).to be_visible
    self
  end

  step 'Validate task is marked as valid'
  def is_valid
    expect(@modal.validation_badge).to be_visible
    self
  end

  step 'Validate task is marked as not valid'
  def is_not_valid
    expect(page).not_to have_css('[data-testid="valid"]')
    self
  end
end
